nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -47,6 +48,15 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
     def first_stage_of_pipeline(self):
         if hasattr(self, "model") and hasattr(self.model, "pre_process"):
             return self.model.pre_process
+        elif (
+            hasattr(self, "model")
+            and isinstance(self.model, Float16Module)
```

Comment:
is this type check necessary? mcore uses a different `Float16Module` class so if you want to do this, I'll need to make it `... and (isinstance(self.model,Float16Module) or isinstance(self.mode, MCoreFloat16Module)) and ...`

will look pretty messy.

Reply:
This `isinstance` check is essentially the same as `hasattr(self.model, 'module')`. It is not necessary, eliminating it still works in my test, and I was just adding it to make the code more readable (the long sequence of `hasattr` is very confusing). 

It really comes to the question that whether checking `hasattr(self.model, 'module'` will consistently work. If provided that Float16Module and MCoreFloat16Module both use .module to save the original network, I think we can remove the `isinstance` check.